Tools/LambdaTestTool-v2/README.md

```diff
@@ -0,0 +1,306 @@
+
+## Installing
+
+The tool is distributed as .NET Global Tool. To install the tool execute the following command:
+
+```
+dotnet tool install -g amazon.lambda.testtool --prerelease
+```
+
+To update the tool run the following command:
+
+```
+dotnet tool update -g amazon.lambda.testtool --prerelease
+```
+
+## Running the Test Tool
+
+### Lambda Emulator Mode
+Use this mode when you want to test Lambda functions directly without API Gateway integration.
+
+```
+# Start Lambda emulator on port 5050
+dotnet lambda-test-tool start --lambda-emulator-port 5050
+```
+
+### API Gateway Emulator Mode
+Use this mode when you want to test Lambda functions through API Gateway endpoints. **Note: Running this mode by itself will not work, you will still need have the lambda runtime client running elsewhere and reference it in the `Endpoint` parameter in the `APIGATEWAY_EMULATOR_ROUTE_CONFIG` env varible (see below [Required Configuration](#required-configuration))** Api gateway mode requires additional configuration through environment variables.
+
+```
+# Start API Gateway emulator on port 5051 in REST mode
+dotnet lambda-test-tool start \
+    --api-gateway-emulator-port 5051 \
+    --api-gateway-emulator-mode Rest
+
+```
+#### Required Configuration
+When running via command line, you must set the environment variable for API Gateway route configuration:
+
+Linux/macOS:
+```bash
+export APIGATEWAY_EMULATOR_ROUTE_CONFIG='{"LambdaResourceName":"AddLambdaFunction","HttpMethod":"Get","Path":"/add/{x}/{y}","Endpoint":"{LAMBDA_RUNTIME_API}"}'
+
+```
+
+Windows (Command Prompt):
+
+```
+set APIGATEWAY_EMULATOR_ROUTE_CONFIG={"LambdaResourceName":"AddLambdaFunction","HttpMethod":"Get","Path":"/add/{x}/{y}","Endpoint":"{LAMBDA_RUNTIME_API}"}
+
+```
+
+Windows (PowerShell):
+
+```
+$env:APIGATEWAY_EMULATOR_ROUTE_CONFIG='{"LambdaResourceName":"AddLambdaFunction","HttpMethod":"Get","Path":"/add/{x}/{y}","Endpoint":"{LAMBDA_RUNTIME_API}"}'
+
+```
+
+Replace `{LAMBDA_RUNTIME_API}` with your Lambda runtime API endpoint (e.g., "http://localhost:5050/AddLambdaFunction" or the endpoint specified in your `AWS_LAMBDA_RUNTIME_API` environment variable).
+
+
+### Combined Mode
+Use this mode when you want to run both Lambda and API Gateway emulators simultaneously.
+
+```
+# Start both emulators
+dotnet lambda-test-tool start \
+    --lambda-emulator-port 5050 \
+    --api-gateway-emulator-port 5051 \
+    --api-gateway-emulator-mode Rest
+```
+
+## Command Line Options
+| Option | Description | Required For |
+|--------|-------------|--------------|
+| `--lambda-emulator-port` | Port for Lambda emulator | Lambda Mode |
+| `--lambda-emulator-host` | Host for Lambda emulator | Lambda Mode |
+| `--api-gateway-emulator-port` | Port for API Gateway | API Gateway Mode |
+| `--api-gateway-emulator-mode` | API Gateway mode (Rest/HttpV1/HttpV2) | API Gateway Mode |
+| `--no-launch-window` | Disable auto-launching web interface | Optional |
+
+
+## API Gateway Configuration
```

Comment:
Can you add a subsection talking about wild card paths like you did in the Aspire README.

Reply:
fixed in https://github.com/aws/aws-lambda-dotnet/pull/1971/commits/696e75800480af5b1b8cd78c35c132d515d33a51